packages/react-dom/src/server/ReactDOMServerFormatConfig.js

```diff
@@ -279,6 +279,7 @@ function encodeHTMLTextNode(text: string): string {
 }
 
 const textSeparator = stringToPrecomputedChunk('<!-- -->');
+let lastPushedText = false;
```

Comment:
You can use `ResponseState` for this instead of module scope. It's a little safer because you can technically call `renderToString` from within `renderToString` I think.

Reply:
good call